### PR TITLE
Tests compatible with 0.5

### PR DESCRIPTION
**Pull request by hiqua**
@ritzdorf FYI. I still have to compare the test outputs to see if they make sense.

### CODE REVIEW

**Comment by ritzdorf** [comment]
Thanks. Let me know once I should take a look.

File: src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json [hunk 1]
```diff
@@ -64,12 +64,12 @@
           106,
           118
         ],
-        "violations": [],
-        "warnings": [
+        "violations": [
```
Comment (hiqua) [inline src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json]: Need to check this.

File: src/test/resources/solidity/locked-money-self-destruct-safe.json [hunk 2]
```diff
@@ -2,64 +2,71 @@
...
+        "warnings": []
       },
       "LockedEther": {
-        "violations": [],
-        "warnings": [],
+        "conflicts": [],
         "safe": [],
-        "conflicts": [
+        "violations": [
           1
```
Comment (hiqua) [inline src/test/resources/solidity/locked-money-self-destruct-safe.json]: LockedEther does not take selfdestruct into account, so need to change that.

File: src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json (hunk 1 above)
Comment (hiqua) [reply src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json]: The assembly code is just slightly reorganized between the two versions so this shouldn't result in a difference in pattern matches.

Comment (ritzdorf) [reply src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json]: Interesting. This is still with the regular decompiler included right?
Could it be that in one of the two cases (either old or new bytecode) the regular decompiler fails and therefore there is a difference?

Comment (hiqua) [reply src/test/resources/solidity/end_to_end_testing_big/swc/transaction_order_dependence/ERC20.json]: Yes in the old case the function decompiler succeeds.

**Comment by hiqua** [comment]
@ritzdorf looks fine.

**Pull request merged**